webapp/app.py

```diff
@@ -73,11 +73,28 @@ def thank_you():
 
 @open_id.after_login
 def after_login(resp):
-    flask.session["macaroon_discharge"] = resp.extensions["macaroon"].discharge
-    flask.session["openid"] = {
-        "identity_url": resp.identity_url,
-        "email": resp.email,
+    root = flask.session["macaroon_root"]
+    discharge = resp.extensions["macaroon"].discharge
+    authorization_header = get_authorization_header(root, discharge)
+    authorization_code = f"root={root} discharge={discharge}"
+    invitation_code = flask.session["invitation_code"]
+    data = {
+        "provider": "usso",
+        "authorization_code": authorization_code,
+        "invitation_code": invitation_code,
     }
+
+    response = requests.post(
+        url=f"{ANBOXCLOUD_API_BASE}1.0/login",
+        headers=authorization_header,
+        json=data,
+    )
+    body = response.json()
+    if response.status_code == 401:
+        return flask.render_template("401.html", error=body["error"])
```

Comment:
`response.raise_for_status` will send an exception that we could get with an error handler that flask provides: https://flask.palletsprojects.com/en/1.1.x/errorhandling/#error-handlers

Reply:
Ok. `raise_for_status` for breaking it, so I was not aware I could use a handler. Thanks

Reply:
But this error is being handled by requests, not flask. I don't seem to be able to intercept it with `app.errorhandler`